websites/R/Rate Your Music/presence.ts

```diff
@@ -0,0 +1,286 @@
+const enum Assets {
+	Logo = "https://i.imgur.com/4hrNGBS.png",
+}
+
+let browsingTimestamp: number;
+
+function updatePresence() {
+	if (!browsingTimestamp) browsingTimestamp = Math.floor(Date.now() / 1000);
+
+	let presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+		startTimestamp: browsingTimestamp,
+	};
+
+	const pages: Record<string, PresenceData> = {
+			"/charts/": {
+				details: "Viewing Chart:",
+			},
+			"/new-music/": {
+				details: "New Releases",
+			},
+			"/genres/": {
+				details: "Genres",
+			},
+			"/lists/": {
+				details: "Lists",
+			},
+			"/subscribe/": {
+				details: "Subscribe",
+			},
+			"/": {
+				details: "Home",
+			},
+		},
+		{ pathname } = document.location;
+
+	//chart page
+	if (pathname.startsWith("/charts/")) {
+		const chartNameElement = document.querySelector(
+			"#page_charts_section_charts_header_chart_name"
+		);
+		if (chartNameElement) {
+			const chartName = chartNameElement.textContent?.trim();
+			if (chartName) {
+				presenceData.details = "Viewing Chart:";
+				presenceData.state = chartName;
+			}
+		}
+	}
+
+	//account page
+	if (pathname.startsWith("/account/")) presenceData.details = "Account";
+
+	//recs page
+	if (pathname.startsWith("/recs/to/") || pathname.startsWith("/recs/from/")) {
```

Comment:
Consider using `else if` or `switch` rather than a bunch of `if`s for performance and better readabilty